File: src/main/java/cloud/requestengine/DownloadStarted.java
package cloud.requestengine;

import se.sics.kompics.address.Address;
import se.sics.kompics.network.Message;

/**
 * 
 * @author Amir Moulavi
 * @date 2011-04-02
 *
 */

public class DownloadStarted extends Message {

	private static final long serialVersionUID = 4603408788313957328L;
	private String requestID;

	public DownloadStarted(Address source, Address destination, String requestID) {
		super(source, destination);
		this.requestID = requestID;
	}

	public String getRequestID() {
		return requestID;
	}

}

File: src/main/java/cloud/requestengine/DownloadStarted.scala
package cloud.requestengine

import se.sics.kompics.network.Message
import se.sics.kompics.address.Address

/**
 * @author Amir Moulavi
 * @date 2011-07-12
 *
 */
class DownloadStarted(val source:Address,
                      val destination:Address,
                      val requestID:String)
  extends Message(source, destination)